File: diitra_web/src/core/documents/registry/DocumentTemplateRegistry.ts
// ═══════════════════════════════════════════════════════════════════
// DIITRA — Document Template Registry (Esquemas de Datos)
//
// RESPONSABILIDAD ÚNICA: Definir la ESTRUCTURA de cada tipo de documento.
//   - Esquema inicial de campos (schema)
//   - Listas colaborativas (lists)
//   - Secciones con sus IDs y labels
//   - Configuración de campos para AgnosticSection (config.fields)
//
// ESTE ARCHIVO NO IMPORTA COMPONENTES REACT.
// Los componentes de React para cada sección están en:
//   → DocumentComponentRegistry.ts
//
// Esto permite que este archivo se use en:
//   - Tests unitarios sin necesidad de React/DOM
//   - Generadores de scripts del backend
//   - Validadores de esquemas
//   - Documentación generada automáticamente
// ═══════════════════════════════════════════════════════════════════

export interface FieldConfig {
    name: string;
    label: string;
    type: 'text' | 'textarea' | 'select' | 'checkbox' | 'number' | 'rich-text' | 'list';
    collaborative: boolean;
    placeholder?: string;
    min?: number;
    max?: number;
    options?: string[];
}

export interface SectionSchema {
    id: string;
    label: string;
    iconName?: string;         // Nombre del ícono de Lucide (ej: 'BookOpen')
    icon_name?: string;
    componentName?: string;
    component_name?: string;
    config?: {
        referenceTemplateCode?: string;
        fields?: FieldConfig[];
    };
}

export interface DocumentSchema {
    title: string;
    subtitle: string;
    schema: Record<string, any>;
    lists: string[];
    sections: SectionSchema[];
}

export const DocumentTemplateRegistry: Record<string, DocumentSchema> = {
    PROTOCOLO_INVESTIGACION: {
        title: "Proyecto de Investigación",
        subtitle: "Formulación del Proyecto de Investigación - ISTPET",
        schema: {
            // Sección 1: Identificación
            Titulo: '',
            IdCarrera: 0,
            IdConvocatoria: 0,
            Periodo: '',
            TiempoEjecucion: '',
            Programa: '',
            GrupoInvestigacionTipo: 'NO',       // 'NO' | 'SI'
            GrupoInvestigacionNombre: '',
            Dominio: '',
            LineaInvestigacion: '',
            SublineaInvestigacion: '',
            TipoInvestigacion: 'APLICADA',       // 'BÁSICA' | 'APLICADA' | 'DESARROLLO EXPERIMENTAL'
            CampoAmplio: '',
            CampoEspecifico: '',
            CampoDetallado: '',
            DirectorProyecto: '',
            FechaPresentacion: '',
            FechaInicio: '',
            FechaFin: '',

            // Sección 2: Investigadores
            Investigadores: [],

            // Sección 3: Especificación Técnica
            Antecedentes: '',
            DescripcionProyecto: '',
            Justificacion: '',
            ObjetivoGeneral: '',
            ObjetivosEspecificos: '',
            ObjetivosDesarrolloSostenible: '',
            MarcoTeorico: '',
            Metodologia: '',
            Evaluacion: '',

            // Sección 4: Recursos, Costo y Financiamiento
            RecursosDisponibles: [],
            RecursosNecesarios: [],
            CostoTotal: 0,
            FinanciamientoIstpet: false,
            FinanciamientoOtrasFuentes: false,
            NombresOtrasFuentes: '',

            // Sección 5: Productos Esperados
            ProductosEsperados: [],

            // Sección 6: Impactos
            Impacto: { social: '', cientifico: '', economico: '', politico: '', ambiental: '', otro: '' },

            // Sección 7: Cronograma
            Cronograma: [],

            // Sección 8: Bibliografía
            Bibliografia: '',

            // Sección 9: Firmas de Responsabilidad
            FirmasResponsabilidad: {
                DirectorNombre: '',
                DirectorCargo: 'Director del Proyecto',
                CoordinadorNombre: '',
                CoordinadorCargo: 'Coordinador de Carrera'
            }
        },
        lists: ['Investigadores', 'RecursosDisponibles', 'RecursosNecesarios', 'Cronograma', 'ProductosEsperados'],
        sections: [
            { id: 'identificacion', label: 'Identificación', iconName: 'FileText', componentName: 'ProjectGeneralSection' },
            { id: 'investigadores', label: 'Investigadores', iconName: 'Users', componentName: 'ResearchersSection' },
            { id: 'especificacion', label: 'Especificación', iconName: 'Layers', componentName: 'ProjectTechnicalSection' },
            { id: 'recursos', label: 'Recursos y Presupuesto', iconName: 'DollarSign', componentName: 'ProjectBudgetSection' },
            { id: 'productos', label: 'Productos Esperados', iconName: 'Package', componentName: 'ExpectedProductsSection' },
            { id: 'impactos', label: 'Impactos', iconName: 'TrendingUp', componentName: 'ImpactsSection' },
            { id: 'cronograma', label: 'Cronograma (Gantt)', iconName: 'Calendar', componentName: 'GanttSection' },
            { id: 'bibliografia', label: 'Bibliografía', iconName: 'BookOpen', componentName: 'AgnosticSection' },
            { id: 'firmas', label: 'Firmas de Responsabilidad', iconName: 'PenTool', componentName: 'SignaturesSection' }
        ]
    },

    PROTOCOLO_INNOVACION: {
        title: "Proyecto de Innovación",
        subtitle: "Formulación del Proyecto de Innovación y Transferencia - ISTPET",
        schema: {
            // Sección 1: Identificación
            Titulo: '',
            CodigoInstitucional: '',
            IdCarrera: 0,
            Carrera: '',
            IdConvocatoria: 0,
            PeriodoConvocatoria: '',
            TiempoEjecucion: '',
            Programa: '',
            GrupoInvestigacionTipo: 'NO',       // 'NO' | 'SI'
            GrupoInvestigacionNombre: '',
            TipoInnovacion: 'PRODUCTO',          // 'PRODUCTO' | 'PROCESO' | 'ORGANIZATIVA' | 'COMERCIAL'
            LineaInvestigacion: '',
            SublineaInvestigacion: '',
            CampoAmplio: '',
            CampoEspecifico: '',
            CampoDetallado: '',
            Alcance: 'INSTITUCIONAL',            // 'INSTITUCIONAL' | 'LOCAL' | 'COMUNITARIO' | 'REGIONAL'
            DirectorProyecto: '',
            FechaPresentacion: '',
            FechaInicioPrevista: '',
            FechaFinPrevista: '',

            // Sección 2: Investigadores
            Investigadores: [],

            // Sección 3: Descripción del Proyecto
            ResumenProyecto: '',
            ObjetivoGeneral: '',
            ObjetivosEspecificos: '',
            Antecedentes: '',
            JustificacionInnovacion: '',
            DescripcionInnovacion: '',
            VinculacionSociedad: '',
            ConvenioAsociado: '',
            EstadoArteConceptual: '',
            MetodologiaAplicacion: '',
            MetodologiaEvaluacion: '',
            Beneficiarios: '',
            ResultadosEsperados: [],
            Viabilidad: '',
            TransferenciaConocimiento: '',

            // Sección 4: Recursos, Costo y Financiamiento
            RecursosDisponibles: [],
            RecursosNecesarios: [],
            CostoTotal: 0,
            FinanciamientoIstpet: false,
            FinanciamientoOtrasFuentes: false,
            NombresOtrasFuentes: '',

            // Sección 5: Impactos
            Impacto: { social: '', cientifico: '', economico: '', ambiental: '', otro: '' },

            // Sección 6: Firmas de Responsabilidad
            FirmasResponsabilidad: {
                DirectorNombre: '',
                DirectorCargo: 'Director del Proyecto de Innovación',
                CoordinadorNombre: '',
                CoordinadorCargo: 'Coordinador de Innovación y Transferencia'
            }
        },
        lists: ['Investigadores', 'RecursosDisponibles', 'RecursosNecesarios', 'ResultadosEsperados'],
        sections: [
            { id: 'identificacion', label: 'Identificación', iconName: 'FileText', componentName: 'InnovationGeneralSection' },
            { id: 'investigadores', label: 'Investigadores', iconName: 'Users', componentName: 'ResearchersSection' },
            { id: 'descripcion', label: 'Descripción del Proyecto', iconName: 'Layers', componentName: 'InnovationDescriptionSection' },
            { id: 'recursos', label: 'Recursos y Presupuesto', iconName: 'DollarSign', componentName: 'ProjectBudgetSection' },
            { id: 'impactos', label: 'Impactos', iconName: 'TrendingUp', componentName: 'ImpactsSection' },
            { id: 'firmas', label: 'Firmas de Responsabilidad', iconName: 'PenTool', componentName: 'SignaturesSection' }
        ]
    },

    RUBRICA_EVALUACION: {
        title: "Rúbrica de Evaluación por Pares",
        subtitle: "Evaluación anónima (Fase 2) — Normativa CACES",
        schema: {
            criterios_evaluados: [],
            ComentariosGenerales: '',
            RecomendacionFinal: ''
        },
        lists: ['criterios_evaluados'],
        sections: []
    },

    INFORME_AVANCE: {
        title: "Informe de Avance de Proyecto",
        subtitle: "Ejecución y Monitoreo (Fase 3)",
        schema: {
            // Sección 1: Bitácora Científica & Actividades Ejecutadas
            ConclusionesParciales: '',
            ActividadesEjecutadas: [],
            ActividadesNoPrevistas: [],
            Obstaculos: [],
            
            // Sección 2: Estado de Ejecución
            EstadoEjecucion: 'EN AVANCE',
            DescripcionFaseActual: '',
            
            // Sección 3: Observaciones y Roles
            ObservacionesDirector: '',
            ObservacionesCoordinador: '',
            
            // Legacy / Compatibilidad
            HitosCompletados: [],
            Evidencias: [],
            PresupuestoEjecutado: [],
        },
        lists: ['ActividadesEjecutadas', 'ActividadesNoPrevistas', 'Obstaculos', 'HitosCompletados', 'Evidencias', 'PresupuestoEjecutado'],
        sections: [
            { id: 'avance_bitacora', label: 'Bitácora Científica', iconName: 'FileText', componentName: 'ProgressLogSection' },
            { id: 'avance_estado', label: 'Estado de Ejecución', iconName: 'CheckCircle', componentName: 'ProgressStateSection' },
            { id: 'avance_observaciones', label: 'Observaciones y Firmas', iconName: 'PenTool', componentName: 'ProgressObservationsSection' }
        ]
    },

    INFORME_FINAL_INVESTIGACION: {
        title: "Informe Final de Investigación",
        subtitle: "Cierre y entrega de resultados - ISTPET",
        schema: {
            Indice: '',
            Resumen: '',
            Introduccion: '',
            Objetivos: '',
            Fundamentos: '',
            Metodos: '',
            Resultados: '',
            Productos: '',
            Impactos: '',
            Transferencia: '',
            InformeFinanciero: '',
            Conclusiones: '',
            Recomendaciones: '',
            Bibliografia: '',
            Anexos: ''
        },
        lists: [],
        sections: [
            { id: 'sec_indice', label: 'Índice y Estructura', iconName: 'List', componentName: 'AgnosticSection' },
            { id: 'sec_resumen', label: 'Resumen Ejecutivo', iconName: 'FileText', componentName: 'AgnosticSection' },
            { id: 'sec_introduccion', label: 'Introducción', iconName: 'BookOpen', componentName: 'AgnosticSection' },
            { id: 'sec_objetivos', label: 'Objetivos y Cumplimiento', iconName: 'Target', componentName: 'AgnosticSection' },
            { id: 'sec_fundamentos', label: 'Fundamentos Teóricos', iconName: 'Layers', componentName: 'AgnosticSection' },
            { id: 'sec_metodos', label: 'Métodos Aplicados', iconName: 'Sliders', componentName: 'AgnosticSection' },
            { id: 'sec_resultados', label: 'Resultados Obtenidos', iconName: 'Activity', componentName: 'AgnosticSection' },
            { id: 'sec_productos', label: 'Productos Alcanzados', iconName: 'Package', componentName: 'AgnosticSection' },
            { id: 'sec_impactos', label: 'Impactos Finales', iconName: 'TrendingUp', componentName: 'AgnosticSection' },
            { id: 'sec_transferencia', label: 'Transferencia', iconName: 'Share2', componentName: 'AgnosticSection' },
            { id: 'sec_informe_financiero', label: 'Informe Financiero', iconName: 'DollarSign', componentName: 'AgnosticSection' },
            { id: 'sec_conclusiones', label: 'Conclusiones', iconName: 'CheckSquare', componentName: 'AgnosticSection' },
            { id: 'sec_recomendaciones', label: 'Recomendaciones', iconName: 'Award', componentName: 'AgnosticSection' },
            { id: 'sec_bibliografia', label: 'Bibliografía Final', iconName: 'Book', componentName: 'AgnosticSection' },
            { id: 'sec_anexos', label: 'Anexos', iconName: 'Paperclip', componentName: 'AgnosticSection' }
        ]
    },

    OFICIO_APROBACION: {
        title: "Formato Oficio de Aprobación de Proyecto",
        subtitle: "Oficio formal emitido por la Coordinación de Investigación para aprobación legal previa a ejecución",
        schema: {
            oficio_numero: "",
            oficio_fecha: "",
            director_titulo: "Tecnólogo/a",
            director_nombre: "",
            director_carrera: "",
            coordinador_nombre: "Ing. Estefani Sánchez Mgtr."
        },
        lists: [],
        sections: [
            {
                id: 'oficio_aprobacion',
                label: 'Oficio de Aprobación',
                iconName: 'FileText',
                config: {
                    fields: [
                        { name: 'oficio_numero', label: 'Número de Oficio', type: 'text', collaborative: false, placeholder: '01-ISTPET-INV-2026' },
                        { name: 'oficio_fecha', label: 'Fecha de Emisión', type: 'text', collaborative: false, placeholder: 'Ej: 27 de marzo de 2026' },
                        { name: 'director_nombre', label: 'Director del Proyecto (Destinatario)', type: 'text', collaborative: false, placeholder: 'Nombre del docente director' },
                        { name: 'director_carrera', label: 'Carrera del Director', type: 'text', collaborative: false, placeholder: 'Tecnología Superior en...' },
                        { name: 'coordinador_nombre', label: 'Coordinador/a de Investigación (Firmante)', type: 'text', collaborative: false, placeholder: 'Ing. Estefani Sánchez Mgtr.' }
                    ]
                }
            }
        ]
    },

    PLAN_APRENDIZAJE: {
        title: "Plan de Aprendizaje de Proyecto",
        subtitle: "Articulación de la Investigación con la Docencia y Prácticas Formativas (APE) — ISTPET",
        schema: {
            // 1. Identificación del Proyecto
            NombreProyecto: '',
            LineaInvestigacion: '',
            SublineaInvestigacion: '',
            Carrera: '',
            DirectorProyecto: '',
            NumeroEstudiantes: 1,
            FechaAprobacion: '',
            FechaTerminacion: '',
            PeriodoAcademico: '',

            // Estudiante activo seleccionado
            EstudianteActivoId: '',

            // 2. Fichas de Estudiantes
            EstudiantesEvaluaciones: [],

            // 6. Firmas de Responsabilidad
            FirmasResponsabilidad: {
                DirectorNombre: '',
                DirectorCargo: 'Director del Proyecto',
                CoordinadorNombre: 'MSc. Christian Castro',
                CoordinadorCargo: 'Coordinador de la Unidad de Investigación'
            }
        },
        lists: ['EstudiantesEvaluaciones'],
        sections: [
            {
                id: 'plan_aprendizaje',
                label: 'Plan de Aprendizaje',
                iconName: 'BookOpen'
            }
        ]
    },

    EVALUACION_PLAN_APRENDIZAJE: {
        title: "Instrumento de Evaluación del Plan de Aprendizaje",
        subtitle: "Evaluación Cualitativa de Articulación con la Docencia (APE) — Coordinación de Investigación ISTPET",
        schema: {
            // 1. Identificación del Proyecto
            NombreProyecto: '',
            LineaInvestigacion: '',
            SublineaInvestigacion: '',
            Carrera: '',
            DirectorProyecto: '',
            NumeroEstudiantes: 1,
            FechaAprobacion: '',
            FechaTerminacion: '',
            PeriodoAcademico: '',

            // Estudiante activo seleccionado
            EstudianteActivoId: '',

            // 2. Fichas de Evaluación por Estudiante
            EstudiantesEvaluaciones: [],

            // 6. Firmas de Responsabilidad
            FirmasResponsabilidad: {
                DirectorNombre: '',
                DirectorCargo: 'Director del Proyecto',
                CoordinadorNombre: 'MSc. Christian Castro',
                CoordinadorCargo: 'Coordinador de la Unidad de Investigación'
            }
        },
        lists: ['EstudiantesEvaluaciones'],
        sections: [
            {
                id: 'evaluacion_plan_aprendizaje',
                label: 'Evaluación Plan de Aprendizaje',
                iconName: 'Award'
            }
        ]
    }
};

File: backend/diitra_infrastructure/Common/Documents/DocumentTemplateRegistry.cs
using Diitra.Domain.Common.Documents;
using System.Collections.Generic;
using System.Linq;
using Diitra.Infrastructure.Common.Documents.Templates.Investigacion;
using Diitra.Infrastructure.Common.Documents.Templates.Innovacion;

namespace Diitra.Infrastructure.Common.Documents
{
    /// <summary>
    /// CATÁLOGO MAESTRO DE PLANTILLAS INSTITUCIONALES (DIITRA Registry)
    /// Este archivo actúa como el índice central. El contenido HTML de cada documento
    /// se encuentra en su propia carpeta bajo /Templates/ para máxima organización.
    /// </summary>
    public static class DocumentTemplateRegistry
    {
        public static DocumentTemplate? GetByCode(string code) => 
            GetSeedTemplates().FirstOrDefault(t => t.Code == code);

        public static IEnumerable<DocumentTemplate> GetSeedTemplates()
        {
            // ══════════════════════════════════════════════════════════════
            // ÁREA: INVESTIGACIÓN
            // ══════════════════════════════════════════════════════════════
            // NOTA: El HTML de cada plantilla vive en su archivo .html correspondiente
            // bajo Templates/{Categoria}/{Nombre}.html. El TemplateFileLoader lo carga
            // automáticamente. El htmlContent aquí es solo un placeholder de arranque;
            // en producción el .html copiado al output tiene prioridad.

            // 1. FORMATO PROYECTO DE INVESTIGACIÓN
            yield return DocumentTemplate.Create(
                code: ProyectoInvestigacionTemplate.CODE,
                name: "1. Formato Proyecto de Investigación",
                description: "Documento oficial para postulación de proyectos I+D+i. Versión de Producción Final v14.0.",
                category: DocumentCategory.Protocolo,
                htmlContent: "<!-- Cargado desde Templates/Investigacion/ProyectoInvestigacion.html -->",
                requiresLopdp: true,
                supportsBlind: true,
                requiresTraceability: true,
                requiresSignature: true,
                collaborativeFields: "[\"programa\", \"grupo_investigacion\", \"dominio\", \"linea_investigacion\", \"sublinea_investigacion\", \"tipo_investigacion\", \"campo_amplio\", \"campo_especifico\", \"campo_detallado\", \"antecedentes\", \"descripcion_proyecto\", \"justificacion\", \"objetivo_general\", \"objetivos_especificos\", \"ods\", \"marco_teorico\", \"metodologia\", \"evaluacion\", \"bibliografia\"]",
                version: 400);

            // 2. FORMATO PROYECTO DE INNOVACIÓN
            yield return DocumentTemplate.Create(
                code: ProyectoInnovacionTemplate.CODE,
                name: "2. Formato Proyecto de Innovación",
                description: "Documento oficial para formulación de proyectos de innovación y transferencia tecnológica - ISTPET.",
                category: DocumentCategory.Protocolo,
                htmlContent: "<!-- Cargado desde Templates/Innovacion/ProyectoInnovacion.html -->",
                requiresLopdp: true,
                supportsBlind: true,
                requiresTraceability: true,
                requiresSignature: true,
                collaborativeFields: "[\"programa\", \"grupo_investigacion\", \"tipo_innovacion\", \"linea_investigacion\", \"sublinea_investigacion\", \"campo_amplio\", \"campo_especifico\", \"campo_detallado\", \"carrera\", \"alcance\", \"resumen_proyecto\", \"objetivo_general\", \"objetivos_especificos\", \"antecedentes\", \"justificacion_innovacion\", \"descripcion_innovacion\", \"vinculacion_sociedad\", \"convenio_asociado\", \"estado_arte_conceptual\", \"metodologia_aplicacion\", \"metodologia_evaluacion\", \"beneficiarios\", \"viabilidad\", \"transferencia_conocimiento\", \"recursos_disponibles\", \"recursos_necesarios\", \"financiamiento\", \"impactos\"]",
                version: 100);

            // 3. PLAN DE APRENDIZAJE DE PROYECTO (ELABORACIÓN DIRECTOR / EQUIPO)
            yield return DocumentTemplate.Create(
                code: "PLAN_APRENDIZAJE",
                name: "Plan de Aprendizaje de Proyecto (Docencia - APE)",
                description: "Planificación de articulación de la investigación con la docencia y prácticas formativas APE - ISTPET.",
                category: DocumentCategory.Protocolo,
                htmlContent: "<!-- Cargado desde Templates/PLAN_APRENDIZAJE.html -->",
                requiresLopdp: true,
                supportsBlind: false,
                requiresTraceability: true,
                requiresSignature: true,
                collaborativeFields: "[\"NombreProyecto\", \"LineaInvestigacion\", \"SublineaInvestigacion\", \"Carrera\", \"DirectorProyecto\", \"NumeroEstudiantes\", \"FechaAprobacion\", \"FechaTerminacion\", \"PeriodoAcademico\", \"NombreEstudiante\", \"ObjetivoGeneral\", \"PrerrequisitosCognitivos\", \"PrerrequisitosProcedimentales\", \"ActividadesPlan\", \"FirmasResponsabilidad\"]",
                version: 200);

            // 4. INSTRUMENTO DE EVALUACIÓN DEL PLAN DE APRENDIZAJE (REVISIÓN ADMINISTRADOR)
            yield return DocumentTemplate.Create(
                code: "EVALUACION_PLAN_APRENDIZAJE",
                name: "Instrumento de Evaluación del Plan de Aprendizaje de los Estudiantes",
                description: "Evaluación cualitativa institucional del plan de aprendizaje realizada por la Coordinación de Investigación - ISTPET.",
                category: DocumentCategory.Protocolo,
                htmlContent: "<!-- Cargado desde Templates/Investigacion/EvaluacionPlanAprendizaje.html -->",
                requiresLopdp: true,
                supportsBlind: false,
                requiresTraceability: true,
                requiresSignature: true,
                collaborativeFields: "[\"NombreProyecto\", \"LineaInvestigacion\", \"SublineaInvestigacion\", \"Carrera\", \"DirectorProyecto\", \"PeriodoAcademico\", \"EstudiantesEvaluaciones\", \"FirmasResponsabilidad\"]",
                version: 200);

            // ══════════════════════════════════════════════════════════════
            // OTRAS ÁREAS (Registro de marcadores de posición)
            // ══════════════════════════════════════════════════════════════

            yield return DocumentTemplate.Create(
                code: "OFICIO_APROBACION",
                name: "Formato Oficio de Aprobación de Proyecto",
                description: "Oficio formal emitido por la Coordinación de Investigación para aprobación legal previa a ejecución.",
                category: DocumentCategory.ActaAprobacion,
                htmlContent: "<!-- Cargado desde Templates/Investigacion/OficioAprobacion.html -->",
                requiresLopdp: true,
                supportsBlind: false,
                requiresTraceability: true,
                requiresSignature: true,
                version: 20);

            yield return DocumentTemplate.Create(
                code: InformeFinalTemplate.CODE,
                name: "Informe Final de Investigación",
                description: "Documento de consolidación de resultados para acreditación CACES.",
                category: DocumentCategory.InformeFinal,
                htmlContent: "<!-- Cargado desde Templates/Investigacion/InformeFinal.html -->",
                requiresLopdp: true,
                supportsBlind: false,
                requiresTraceability: true,
                requiresSignature: true,
                collaborativeFields: "[\"Indice\", \"sec_indice\", \"Resumen\", \"sec_resumen\", \"resumen_ejecutivo\", \"Introduccion\", \"sec_introduccion\", \"Objetivos\", \"sec_objetivos\", \"cumplimiento_objetivos\", \"Fundamentos\", \"sec_fundamentos\", \"Metodos\", \"sec_metodos\", \"Resultados\", \"sec_resultados\", \"Productos\", \"sec_productos\", \"Impactos\", \"sec_impactos\", \"impacto_final\", \"Transferencia\", \"sec_transferencia\", \"transferencia_conocimiento\", \"InformeFinanciero\", \"sec_informe_financiero\", \"Conclusiones\", \"sec_conclusiones\", \"Recomendaciones\", \"sec_recomendaciones\", \"Bibliografia\", \"sec_bibliografia\", \"bibliografia_final\", \"Anexos\", \"sec_anexos\"]",
                version: 20);

            yield return DocumentTemplate.Create(
                code: RubricaEvaluacionTemplate.CODE,
                name: "Rúbrica de Evaluación por Pares",
                description: "Revisión doble ciego (Fase 2) — Normativa CACES.",
                category: DocumentCategory.Protocolo,
                htmlContent: "<!-- Cargado desde Templates/Investigacion/RubricaEvaluacion.html -->",
                requiresLopdp: true,
                supportsBlind: true,
                requiresTraceability: true,
                requiresSignature: true,
                collaborativeFields: "[\"Pertinencia\", \"Metodologia\", \"Viabilidad\", \"Impacto\", \"ComentariosGenerales\", \"RecomendacionFinal\"]",
                version: 20);

            yield return DocumentTemplate.Create(
                code: InformeAvanceTemplate.CODE,
                name: "Informe de Avance de Proyecto",
                description: "Ejecución y Monitoreo (Fase 3).",
                category: DocumentCategory.InformeAvance,
                htmlContent: "<!-- Cargado desde Templates/Investigacion/InformeAvance.html -->",
                requiresLopdp: true,
                supportsBlind: false,
                requiresTraceability: true,
                requiresSignature: true,
                collaborativeFields: "[\"ActividadesEjecutadas\", \"ActividadesNoPrevistas\", \"Obstaculos\", \"ConclusionesParciales\", \"EstadoEjecucion\", \"DescripcionFaseActual\", \"ObservacionesDirector\", \"ObservacionesCoordinador\", \"HitosCompletados\", \"Evidencias\", \"PresupuestoEjecutado\"]",
                version: 20);

            yield return DocumentTemplate.Create(
                code: ReporteAnaliticasTemplate.CODE,
                name: "Reporte de Analíticas de Investigación e Innovación",
                description: "Reporte directivo con indicadores KPI, cumplimiento CACES y portafolio de proyectos para acreditación institucional.",
                category: DocumentCategory.ReporteAnaliticas,
                htmlContent: "<!-- Cargado desde Templates/Investigacion/ReporteAnaliticas.html -->",
                requiresLopdp: true,
                supportsBlind: false,
                requiresTraceability: true,
                requiresSignature: false,
                version: 20);

            yield return DocumentTemplate.Create(
                code: "DICTAMEN_ARBITRAJE",
                name: "Acta de Dictamen de Arbitraje",
                description: "Documento oficial CACES del resultado de la evaluación por pares doble ciego. Firmable digitalmente por el Director de Investigación.",
                category: DocumentCategory.DictamenArbitraje,
                htmlContent: "<!-- Cargado desde Templates/Investigacion/DictamenArbitraje.html -->",
                requiresLopdp: false,
                supportsBlind: true,
                requiresTraceability: true,
                requiresSignature: true,
                version: 20);

            // ══════════════════════════════════════════════════════════════
            // CERTIFICADOS DE COMPLETACIÓN Y PARTICIPACIÓN
            // ══════════════════════════════════════════════════════════════
            yield return DocumentTemplate.Create(
                code: "CERTIFICADO_COMPLETACION",
                name: "Certificado de Completación de Proyecto",
                description: "Certificado oficial otorgado a estudiantes, docentes y directores al finalizar con éxito un proyecto de investigación.",
                category: DocumentCategory.CertificadoCompletacion,
                htmlContent: "<!-- Cargado desde Templates/Investigacion/CertificadoCompletacion.html -->",
                requiresLopdp: true,
                supportsBlind: false,
                requiresTraceability: true,
                requiresSignature: true,
                version: 10);

            yield return DocumentTemplate.Create(
                code: "CERTIFICADO_PARTICIPACION_GRUPO",
                name: "Certificado de Participación de Grupo",
                description: "Certificado de reconocimiento por pertenecer a un Grupo o Semillero de Investigación.",
                category: DocumentCategory.CertificadoGrupo,
                htmlContent: "<!-- Cargado desde Templates/Investigacion/CertificadoGrupo.html -->",
                requiresLopdp: true,
                supportsBlind: false,
                requiresTraceability: true,
                requiresSignature: true,
                version: 10);

            // ══════════════════════════════════════════════════════════════
            // GRUPOS Y SEMILLEROS DE INVESTIGACIÓN
            // ══════════════════════════════════════════════════════════════
            yield return DocumentTemplate.Create(
                code: "PROPUESTA_GRUPO_INVESTIGACION",
                name: "Formato Propuesta de Creación de Grupo de Investigación",
                description: "Documento oficial para formulación, postulación y trámite de conformación de grupos y semilleros de investigación - ISTPET.",
                category: DocumentCategory.PropuestaGrupoInvestigacion,
                htmlContent: "<!-- Cargado desde Templates/Investigacion/PropuestaGrupoInvestigacion.html -->",
                requiresLopdp: true,
                supportsBlind: false,
                requiresTraceability: true,
                requiresSignature: true,
                collaborativeFields: "[\"nombre_grupo\", \"siglas\", \"tipo_grupo\", \"dominio\", \"lineas_investigacion\", \"carreras_vinculadas\", \"mision\", \"vision\", \"objetivo_general\", \"coordinador_nombre\", \"coordinador_cedula\", \"coordinador_email\", \"coordinador_telefono\", \"miembros_docentes\", \"miembros_estudiantes\"]",
                version: 10);

            // Nota: Para agregar una nueva plantilla:
            //   1. Crear el .html en Templates/{Categoria}/{NombreArchivo}.html
            //   2. Agregar el CODE al TemplateFileLoader.ResolveFilePath()
            //   3. Registrar aquí con htmlContent vacío
        }

    }
}
